superset/dashboards/commands/importers/v0.py

```diff
@@ -0,0 +1,336 @@
+    # are converted to filter_scopes
+    # but dashboard create from import may still have old dashboard filter metadata
+    # here we convert them to new filter_scopes metadata first
+    filter_scopes = {}
+    if (
+        "filter_immune_slices" in i_params_dict
+        or "filter_immune_slice_fields" in i_params_dict
+    ):
+        filter_scopes = convert_filter_scopes(old_json_metadata, slices)
+
+    if "filter_scopes" in i_params_dict:
+        filter_scopes = old_json_metadata.get("filter_scopes")
+
+    # then replace old slice id to new slice id:
+    if filter_scopes:
+        new_filter_scopes = copy_filter_scopes(
+            old_to_new_slc_id_dict=old_to_new_slc_id_dict,
+            old_filter_scopes=filter_scopes,
+        )
+
+    # override the dashboard
+    existing_dashboard = None
+    for dash in session.query(Dashboard).all():
+        if (
+            "remote_id" in dash.params_dict
+            and dash.params_dict["remote_id"] == dashboard_to_import.id
+        ):
+            existing_dashboard = dash
+
+    dashboard_to_import = dashboard_to_import.copy()
+    dashboard_to_import.id = None
+    dashboard_to_import.reset_ownership()
+    # position_json can be empty for dashboards
+    # with charts added from chart-edit page and without re-arranging
+    if dashboard_to_import.position_json:
+        alter_positions(dashboard_to_import, old_to_new_slc_id_dict)
+    dashboard_to_import.alter_params(import_time=import_time)
+    dashboard_to_import.remove_params(param_to_remove="filter_immune_slices")
+    dashboard_to_import.remove_params(param_to_remove="filter_immune_slice_fields")
+    if new_filter_scopes:
+        dashboard_to_import.alter_params(filter_scopes=new_filter_scopes)
+    if new_expanded_slices:
+        dashboard_to_import.alter_params(expanded_slices=new_expanded_slices)
+    if new_timed_refresh_immune_slices:
+        dashboard_to_import.alter_params(
+            timed_refresh_immune_slices=new_timed_refresh_immune_slices
+        )
+
+    new_slices = (
+        session.query(Slice).filter(Slice.id.in_(old_to_new_slc_id_dict.values())).all()
+    )
+
+    if existing_dashboard:
+        existing_dashboard.override(dashboard_to_import)
+        existing_dashboard.slices = new_slices
+        session.flush()
+        return existing_dashboard.id
+
+    dashboard_to_import.slices = new_slices
+    session.add(dashboard_to_import)
+    session.flush()
+    return dashboard_to_import.id  # type: ignore
+
+
+def decode_dashboards(  # pylint: disable=too-many-return-statements
+    o: Dict[str, Any]
+) -> Any:
+    """
+    Function to be passed into json.loads obj_hook parameter
+    Recreates the dashboard object from a json representation.
+    """
+    from superset.connectors.druid.models import (
+        DruidCluster,
+        DruidColumn,
+        DruidDatasource,
+        DruidMetric,
+    )
+
+    if "__Dashboard__" in o:
```

Comment:
we should have a more definitive way on knowing what this object type is, can we add a todo here to clean this up upstream later

Reply:
Good point, we should definitely improve this. I'll add a TODO and do a cleanup later.